ps_socialfollow.php

```diff
@@ -283,13 +320,27 @@ protected function updateFields()
     {
         $validator = Validation::createValidator();
         $constraints = [new Url()];
-
+        $config = [];
+        $errors = [];
         foreach (static::SOCIAL_NETWORKS as $social) {
-            $value = Tools::getValue('blocksocial_' . $social, '');
-            $violations = $validator->validate($value, $constraints);
-            if (0 === count($violations)) {
-                Configuration::updateValue('BLOCKSOCIAL_' . strtoupper($social), $value);
+            foreach ($this->context->controller->getLanguages() as $lang) {
+                $id_lang = $lang['id_lang'];
+                $config[$social][$id_lang] = Tools::getValue('blocksocial_' . $social . '_' . $id_lang, '');
+                $violations = $validator->validate($config[$social][$id_lang], $constraints);
+                if (count($violations)) {
+                    $errors[] = $this->trans('Invalid URL', [], 'Admin.Notifications.Error') . ': ' . $config[$social][$id_lang];
+                }
+            }
+
+        }
+        if (empty($errors)) {
+            foreach (static::SOCIAL_NETWORKS as $social) {
```

Comment:
Nice-to-have: have a private method to ease the reading, even if it's quite basic
```
public function saveAllSocialNetworks(array $newConfiguration)
{
    foreach (static::SOCIAL_NETWORKS as $socialNetwork) {
        Configuration::updateValue($this->buildBlockIdentifier($socialNetwork), $newConfiguration[$socialNetwork]);
    }
}
```

Reply:
A private method requires to jump around in the code to read it. I feel like nested function calls should be avoided, when they don't go in the direction of either DRY or separation of concerns.

Reply:
> A private method requires to jump around in the code to read it

IMO, this happens only if the private method signature is not clear enough to explain what's going inside. If we need to open a private method to read the code inside, the method was name, the parameter names, and the phpDoc are not good enough.

Private methods is useful to hide some complexity to the developers reading a function. If the developer is interested in the implementation, indeed he will have to jump around the code. If he's rather reading a function and just interested in "what's happening here", then he might be happy with a private method that, through its name, tell him what the code is doing without having him to analyze the logic. Private methods carry meaning.

Reply:
Example:

```
public function processCustomer(Customer $customer)
{
   ... do things ...
   $this->flushCustomerCache();
   ... do things ...
   $this->flushCustomerCache();
}
```
In the above example; you don't need to open the `flushCustomerCache` method to know what it does. But you need to open it if you are interested in _how_ it does it. So `flushCustomerCache` is IMO a useful private method as it eases the reading of `processCustomer`